### ISSUE TITLE
Restarting sysinit after rebuild

### ISSUE THREAD

**Issue by alsvartr** [NixOS/nix#5318]
![photo_2021-10-02_10-57-41](https://user-images.githubusercontent.com/205880/135708702-6ead3b0d-3eae-4958-b640-e1b2586e49a5.jpg)
nixos-rebuild test restarting sysinit leading to system shutdown so I can't update.
nixos-rebuild switch leads to broken system (can't boot on stage2).

nix-env --version output
nix-env (Nix) 2.3.15

**Comment by bjornfor** [comment]
Hi, this issue belongs in nixpkgs, please repost there. (Can you try "nixos-rebuild boot"?)

**Comment by alsvartr** [comment]
> Hi, this issue belongs in nixpkgs, please repost there.

No, I think this is nixos issue because upgrade is leading to unbootable system. It's not a package issue, I'm doing a simple standard operation - updating the system.

> (Can you try "nixos-rebuild boot"?)

Yes, won't work.

**Comment by bjornfor** [comment]
NixOS issues go in https://github.com/nixos/nixpkgs/, because NixOS sources are located there. This repo is for the Nix package manager itself (cli + daemon). Nix doesn't decide how NixOS does its upgrades / activation. Nix only builds packages. What you're looking at is a NixOS package/program doing something wrong/unexpected related to upgrading the system.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
